aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2997,6 +2997,18 @@ static void apply_svd(Tensor& self, Tensor& U, Tensor& S, Tensor& VT,
     // Compute S, U (optionally) and VT (optionally)
     lapackSvd<scalar_t, value_t>(jobz, m, n, self_working_ptr, lda,
                         S_working_ptr, U_working_ptr, lda, VT_working_ptr, ldvt, work_data, lwork, rwork_data, iwork_data, &info);
+
+// Reference LAPACK changed `info` behavior for inputs with non-finite values
+// Previously, it would return `info` > 0, but now it returns `info` = -4
+// Making the behaviour different from MKL
+// Here we check for the case where `info` is -4 and set it to a positive number
+// This will give the same error message as with MKL
+#if AT_BUILD_WITH_OPENBLAS()
```

Comment:
Should this be unconditional?

According to the LAPACK documentation in 3.10, this is expected not only for OPENBLAS, but for all LAPACK-complaint backends. See
https://www.netlib.org/lapack/explore-html/d3/da8/group__complex16_g_esing_gaccb06ed106ce18814ad7069dcb43aa27.html

Reply:
PyTorch's build system doesn't allow mixing different implementations of BLAS and LAPACK. Both must be from OpenBLAS, or MKL, or Accelerate/vecLib, ... .
MKL 2021.4 is not compliant with the latest Reference LAPACK implementation.
https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-fortran/top/lapack-routines/lapack-least-squares-and-eigenvalue-problem/lapack-least-squares-eigenvalue-problem-driver/singular-value-decomposition-lapack-driver/gesdd.html?wapkw=gesdd?wapkw=gesdd
There is no documentation for Apple's Accelerate/vecLib.

There are really only two used BLAS&LAPACK implementations for CPU computing which are MKL and OpenBLAS. We can safely ignore other options until someone actually complains.

Other implementations allowed by `pytorch/cmake/Modules/FindLAPACK.cmake`:
* GotoBLAS2 - not maintained anymore
* FLAME - can't find documentation, but [the code](https://github.com/flame/libflame/blob/master/src/lapack/dec/svd/front/flamec/FLA_Svd.c) didn't change from 2014 
* AMD Core Math Library (ACML) - not maintained anymore

Switching the `-4` value of `info` for any LAPACK implementation shouldn't hurt anything.
We could also specialize SVD's error with `info = -4` to `Sorry, the input contained non-finite values and the computation cannot be completed`, but we don't have the same info from MKL to raise the same error.